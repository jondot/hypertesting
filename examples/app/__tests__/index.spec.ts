import path from 'path'
import hypertest from '../../../src/index'
import app from '../index'

const test = hypertest(() => Promise.resolve({ app, closeApp: () => {} }), {
  expect
})
describe('app', () => {
  it('hypertest', async () => {
    await test(path.join(__dirname, 'requests'))
  })
})
