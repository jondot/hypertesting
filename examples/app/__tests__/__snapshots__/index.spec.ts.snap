// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app hypertest: requests/api.yaml custom scrub: with a custom scrub 1`] = `
  GET scrubbed

  Object {
    "request": Object {
      "body": undefined,
      "headers": Object {
        "user-agent": "node-superagent/3.8.3",
      },
    },
    "response": Object {
      "body": "{\\"hello\\":\\"scrubbed\\"}",
      "headers": Object {
        "connection": "scrubbed",
        "content-length": "17",
        "content-type": "application/json; charset=utf-8",
        "date": "scrubbed",
        "etag": "scrubbed",
        "x-powered-by": "Express",
      },
      "status": 200,
    },
  }
`;

exports[`app hypertest: requests/api.yaml my-first-request: just a normal GET request 1`] = `
  GET scrubbed

  Object {
    "request": Object {
      "body": undefined,
      "headers": Object {
        "user-agent": "node-superagent/3.8.3",
      },
    },
    "response": Object {
      "body": "{\\"hello\\":\\"world\\"}",
      "headers": Object {
        "connection": "close",
        "content-length": "17",
        "content-type": "application/json; charset=utf-8",
        "date": "scrubbed",
        "etag": "scrubbed",
        "x-powered-by": "Express",
      },
      "status": 200,
    },
  }
`;
